#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

			     BASH PATCH REPORT
			     =================

Bash-Release: 3.1
Patch-ID: bash31-002

Bug-Reported-by: vapier@gentoo.org
Bug-Reference-ID: <20051210223218.GD3324@toucan.gentoo.org>
Bug-Reference-URL:http://lists.gnu.org/archive/html/bug-bash/2005-12/msg00021.html

Bug-Description:

This corrects several omissions in the bash documentation:  It adds the new
options to `ulimit', the exact expansions for `case' patterns, clarification
of the language concerning the return value of `[[', and updated version
information.

Patch:

*** ../bash-3.1/doc/bashref.texi	Mon Oct  3 15:07:21 2005
--- doc/bashref.texi	Fri Dec 30 10:50:39 2005
***************
*** 962,967 ****
  is enabled, the match is performed without regard to the case
  of alphabetic characters.
! The return value is 0 if the string matches or does not match
! the pattern, respectively, and 1 otherwise.
  Any part of the pattern may be quoted to force it to be matched as a
  string.
--- 962,967 ----
  is enabled, the match is performed without regard to the case
  of alphabetic characters.
! The return value is 0 if the string matches (@samp{==}) or does not
! match (@samp{!=})the pattern, and 1 otherwise.
  Any part of the pattern may be quoted to force it to be matched as a
  string.
***************
*** 2599,2603 ****
  or inconvenient to obtain with separate utilities.
  
! This section briefly the builtins which Bash inherits from
  the Bourne Shell, as well as the builtin commands which are unique
  to or have been extended in Bash.
--- 2597,2601 ----
  or inconvenient to obtain with separate utilities.
  
! This section briefly describes the builtins which Bash inherits from
  the Bourne Shell, as well as the builtin commands which are unique
  to or have been extended in Bash.
***************
*** 3834,3838 ****
  @btindex ulimit
  @example
! ulimit [-acdflmnpstuvSH] [@var{limit}]
  @end example
  @code{ulimit} provides control over the resources available to processes
--- 3834,3838 ----
  @btindex ulimit
  @example
! ulimit [-acdfilmnpqstuvxSH] [@var{limit}]
  @end example
  @code{ulimit} provides control over the resources available to processes
***************
*** 3858,3861 ****
--- 3858,3864 ----
  The maximum size of files created by the shell.
  
+ @item -i
+ The maximum number of pending signals.
+ 
  @item -l
  The maximum size that may be locked into memory.
***************
*** 3870,3873 ****
--- 3873,3879 ----
  The pipe buffer size.
  
+ @item -q
+ The maximum number of bytes in POSIX message queues.
+ 
  @item -s
  The maximum stack size.
***************
*** 3882,3885 ****
--- 3888,3894 ----
  The maximum amount of virtual memory available to the process.
  
+ @item -x
+ The maximum number of file locks.
+ 
  @end table
  
***************
*** 4090,4095 ****
  
  @item -x
! Print a trace of simple commands, \fBfor\fP commands, \fBcase\fP
! commands, \fBselect\fP commands, and arithmetic \fBfor\fP commands
  and their arguments or associated word lists after they are
  expanded and before they are executed.  The value of the @env{PS4}
--- 4103,4108 ----
  
  @item -x
! Print a trace of simple commands, @code{for} commands, @code{case}
! commands, @code{select} commands, and arithmetic @code{for} commands
  and their arguments or associated word lists after they are
  expanded and before they are executed.  The value of the @env{PS4}
*** ../bash-3.1/doc/version.texi	Tue Sep 20 14:52:56 2005
--- doc/version.texi	Fri Dec 30 10:50:58 2005
***************
*** 3,10 ****
  @end ignore
  
! @set LASTCHANGE Mon Sep  5 11:47:04 EDT 2005
  
! @set EDITION 3.1-beta1
! @set VERSION 3.1-beta1
! @set UPDATED 5 September 2005
! @set UPDATED-MONTH September 2005
--- 3,10 ----
  @end ignore
  
! @set LASTCHANGE Fri Dec 30 10:50:51 EST 2005
  
! @set EDITION 3.1
! @set VERSION 3.1
! @set UPDATED 30 December 2005
! @set UPDATED-MONTH December 2005
